akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessCoordinator.scala

```diff
@@ -0,0 +1,211 @@
+import akka.cluster.ddata.typed.scaladsl.ReplicatorMessageAdapter
+import akka.cluster.sharding.typed.ChangeNumberOfProcesses
+import akka.cluster.sharding.typed.ShardedDaemonProcessCommand
+import akka.pattern.StatusReply
+
+import java.time.Instant
+import scala.concurrent.duration.DurationInt
+
+/**
+ * INTERNAL API
+ *
+ * A ShardedDaemonProcessCoordinator manages dynamic scaling of one setup of ShardedDaemonProcess
+ */
+private[akka] object ShardedDaemonProcessCoordinator {
+
+  private final case class ScaleState(revision: Int, numberOfProcesses: Int, completed: Boolean, started: Instant)
+
+  private type Register = LWWRegister[ScaleState]
+  private def keyFor(name: String) = LWWRegisterKey[ScaleState](name)
+  private trait InternalCommand extends ShardedDaemonProcessCommand
+
+//   private case class InternalSubscribeResponse(chg: Replicator.SubscribeResponse[Register]) extends InternalCommand
+
+  private case class InternalGetResponse(rsp: Replicator.GetResponse[Register]) extends InternalCommand
+
+  private case class InternalUpdateResponse(rsp: Replicator.UpdateResponse[Register]) extends InternalCommand
+
+  def apply(initialNumberOfProcesses: Int, daemonProcessName: String): Behavior[ShardedDaemonProcessCommand] =
+    Behaviors.setup { context =>
+      context.log.debug("ShardedDaemonProcessCoordinator for [{}] starting", daemonProcessName)
+      val key = keyFor(daemonProcessName)
+      val ddata = DistributedData(context.system)
+      val selfUniqueAddress = ddata.selfUniqueAddress
+
+      DistributedData.withReplicatorMessageAdapter[ShardedDaemonProcessCommand, Register] { replicatorAdapter =>
+        new ShardedDaemonProcessCoordinator(
+          context,
+          daemonProcessName,
+          initialNumberOfProcesses,
+          key,
+          selfUniqueAddress,
+          replicatorAdapter).start()
+      }
+    }
+
+}
+
+private class ShardedDaemonProcessCoordinator(
+    context: ActorContext[ShardedDaemonProcessCommand],
+    daemonProcessName: String,
+    initialNumberOfProcesses: Int,
+    key: LWWRegisterKey[ShardedDaemonProcessCoordinator.ScaleState],
+    selfUniqueAddress: SelfUniqueAddress,
+    replicatorAdapter: ReplicatorMessageAdapter[ShardedDaemonProcessCommand, ShardedDaemonProcessCoordinator.Register]) {
+  import ShardedDaemonProcessCoordinator._
+  private val started = Instant.now()
+  // FIXME write timeouts from config
+  private val writeTimeout = 5.seconds
+  private def initialState =
+    LWWRegister(
+      selfUniqueAddress,
+      ScaleState(
+        revision = 0,
+        numberOfProcesses = initialNumberOfProcesses,
+        completed = true,
+        // not quite correct but also not important
+        started = started))
+
+  // FIXME stash or deny new requests if rescale in progress?
+  def start(): Behavior[ShardedDaemonProcessCommand] = {
+    replicatorAdapter.askGet(
+      replyTo => Replicator.Get(key, Replicator.ReadLocal, replyTo),
```

Comment:
should use some kind of read and write majority to be safe

Reply:
I did a write all to know it reaches all pingers, so a local read would be ok, but I'm not convinced that's a good solution for pausing the pingers anyway.

Reply:
Went with read/write majority and explicit ping protocol in [d608bee](https://github.com/akka/akka/pull/31844/commits/d608bee914c17f59d688111263e58b77c0a601b9)